Seminars/Sem.04/Pract.04/Task02.cpp

```diff
@@ -0,0 +1,224 @@
+#include <iostream>
+#include <fstream>
+using std::cin;
+using std::cout;
+using std::endl;
+
+
+const int MAX_NAME_LENGTH = 20;
+const int CONDITION_LENGTH = 50;
+const int  NUMBER_OF_TASKS = 10;
+const int NUM_TASKS = 3;
+const int MIN_PASS_GRADE = 60;
+
+class Task {
+private:
+	char name[MAX_NAME_LENGTH];
+	char condition[CONDITION_LENGTH];
+	int points;
+public:
+	Task():name{}, condition{}, points{} {}
+	Task(const char* name, const char* condition, int points) {
+		setName(name);
+		setCondition(condition);
+		setPoints(points);
+	}
+	const char* getName()const {
+		return name;
+	}
+	const char* getCondition()const {
+		return condition;
+	}
+	int getPoints()const {
+		return points;
+	}
+
+	void setName(const char name[]) {
+		int index = 0;
+		while (index != '\0') {
+			this->name[index] = name[index];
+			index++;
+		}
+		this->name[index] = '\0';
+	}
+	void setCondition(const char condition[]) {
+		int index = 0;
+		while (index != '\0') {
+			this->condition[index] = condition[index];
+			index++;
+		}
+		this->condition[index] = '\0';
+	}
+	void setPoints(int points) {
+		if (points >= 0) {
+			this->points = points;
+		}
+	}
+	void changeName(const char name[]) {
+		setName(name);
+	}
+	void changeCondition(const char condition[]) {
+		setCondition(condition);
+
+	}
+	void changePoints(int points) {
+		setPoints(points);
+	}
+};
+class Exam {
+private:
+	Task tasks[NUMBER_OF_TASKS];
+	int numOfTasks;
+	int minPoints;
+
+public:
+	Exam(const Task tasks[], int minPoints, int numOfTasks) {
+		setTasks(tasks, numOfTasks);
+		setMinPoints(minPoints);
+
+	}
+
+	void setTasks(const Task tasks[], int numOfTasks){
+
+		if (numOfTasks >= 0) {
+			this ->numOfTasks = numOfTasks;
+		}
+		
+		for (int i = 0; i < numOfTasks; i++) {
+			this->tasks[i] = tasks[i];
+		}
+
+     }
+	void setMinPoints(int minPoints) {
```

Comment:
You use this in the constructor and do not initialise minPoints if the argument is negative